lib/AST/Decl.cpp

```diff
@@ -2610,6 +2610,16 @@ ArtificialMainKind ClassDecl::getArtificialMainKind() const {
   llvm_unreachable("class has no @ApplicationMain attr?!");
 }
 
+bool ClassDecl::isXCTestCaseSubclass() const {
+  ASTContext &ctx = getASTContext();
+  for (auto D = getSuperclassDecl(); D; D = D->getSuperclassDecl()) {
+    if (D->getName() == ctx.Id_XCTestCase)
```

Comment:
We don't want to check `hasClangDecl()`, because corelibs XCTest also defines this, and we use this function to identify test cases there as well.